include/internal/param_names.h.in

```diff
@@ -0,0 +1,47 @@
+/*
+ * {- join("\n * ", @autowarntext) -}
+ *
+ * Copyright 2023 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+{-
+use OpenSSL::paramnames qw(generate_internal_macros);
+-}
+
+#define MAX_PARAM_COUNT     100
+
+typedef struct param_idx {
+    OSSL_PARAM *param;
+    int type;
+} OSSL_PARAM_IDX[MAX_PARAM_COUNT];
```

Comment:
I was kind of thinking of having this array length be specified at each function using these functions, so that it can be long enough as needed for that specific function. It'd be similar to the pattern we often use to construct OSSL_PARAMs in the first place.

However, I guess the issue is the OSSL_PARAM semantics are to ignore unrecognised parameters. So theoretically a client could stuff in an infinite number of irrelevant parameters. That's certainly a problem.

How would the original PR handle multiple instances of the same parameter, for that matter? I assume callers aren't required to pass multiple instances of the same parameter consecutively. Hmm...